bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,901 @@
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.openhab.binding.verisure.internal.model.VerisureAlarms;
+import org.openhab.binding.verisure.internal.model.VerisureBroadbandConnections;
+import org.openhab.binding.verisure.internal.model.VerisureClimates;
+import org.openhab.binding.verisure.internal.model.VerisureDoorWindows;
+import org.openhab.binding.verisure.internal.model.VerisureInstallations;
+import org.openhab.binding.verisure.internal.model.VerisureInstallations.Owainstallation;
+import org.openhab.binding.verisure.internal.model.VerisureMiceDetection;
+import org.openhab.binding.verisure.internal.model.VerisureMiceDetection.Mouse;
+import org.openhab.binding.verisure.internal.model.VerisureSmartLock;
+import org.openhab.binding.verisure.internal.model.VerisureSmartLocks;
+import org.openhab.binding.verisure.internal.model.VerisureSmartPlugs;
+import org.openhab.binding.verisure.internal.model.VerisureThing;
+import org.openhab.binding.verisure.internal.model.VerisureUserPresences;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * This class performs the communication with Verisure My Pages.
+ *
+ * @author Jarle Hjortland - Initial contribution
+ * @author Jan Gustafsson - Re-design and support for several sites and update to new Verisure API
+ *
+ */
+@NonNullByDefault
+public class VerisureSession {
+
+    @NonNullByDefault
+    private final class VerisureInstallation {
+        private @Nullable String installationName;
+        private @Nullable BigDecimal installationId;
+        private @Nullable String pinCode;
+
+        public @Nullable String getPinCode() {
+            return pinCode;
+        }
+
+        public void setPinCode(@Nullable String pinCode) {
+            this.pinCode = pinCode;
+        }
+
+        public VerisureInstallation() {
+        }
+
+        public @Nullable BigDecimal getInstallationId() {
+            return installationId;
+        }
+
+        public @Nullable String getInstallationName() {
+            return installationName;
+        }
+
+        public void setInstallationId(@Nullable BigDecimal installationId) {
+            this.installationId = installationId;
+        }
+
+        public void setInstallationName(@Nullable String installationName) {
+            this.installationName = installationName;
+        }
+    }
+
+    private final HashMap<String, VerisureThing> verisureThings = new HashMap<String, VerisureThing>();
+    private final Logger logger = LoggerFactory.getLogger(VerisureSession.class);
+    private final Gson gson = new GsonBuilder().create();
+    private final List<DeviceStatusListener> deviceStatusListeners = new CopyOnWriteArrayList<>();
+    private final Hashtable<@Nullable BigDecimal, @Nullable VerisureInstallation> verisureInstallations = new Hashtable<@Nullable BigDecimal, @Nullable VerisureInstallation>();
```

Comment:
Why are you using a Hashtable? Please use a HashMap instead. Also, why are you allowing nullable keys?

Reply:
Changed to HashMap and removed nullable.